orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/expressions/functions/ArtifactExpressionFunctionProviderSpec.groovy

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.expressions.functions
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact
+import com.netflix.spinnaker.kork.artifacts.model.ExpectedArtifact
+import com.netflix.spinnaker.kork.expressions.SpelHelperFunctionException
+import com.netflix.spinnaker.orca.pipeline.model.DefaultTrigger
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static com.netflix.spinnaker.orca.pipeline.expressions.functions.ArtifactExpressionFunctionProvider.triggerResolvedArtifact
+import static com.netflix.spinnaker.orca.pipeline.expressions.functions.ArtifactExpressionFunctionProvider.triggerResolvedArtifactByType
+import static com.netflix.spinnaker.orca.test.model.ExecutionBuilder.pipeline
+
+
+class ArtifactExpressionFunctionProviderSpec extends Specification {
+  @Shared
+  def pipeline1 = pipeline {
+    def matchArtifact1 = new Artifact(type: "docker/image", "name": "covfefe")
```

Comment:
nit: maybe not `covfefe` - don't want to (add) to already existing political references in the codebase + it's easier to read if you know what it is, e.g.
```suggestion
    def matchArtifact1 = new Artifact(type: "docker/image", "name": "artifactName1")
```

Reply:
(and similar below)